code/modules/clothing/head/helmet.dm

```diff
@@ -115,3 +139,147 @@ obj/item/clothing/head/helmet/bluetaghelm
 	armor = list(melee = 30, bullet = 10, laser = 20,energy = 10, bomb = 20, bio = 0, rad = 0)
 	// Offer about the same protection as a hardhat.
 	flags_inv = HIDEEARS|HIDEEYES
+	canAttachCam =  0
+
+//LightToggle
+
+/obj/item/clothing/head/helmet/update_icon()
+
+	var/state = "[initial(icon_state)]"
+	if(helmetCam)
+		state += "-cam" //"helmet-cam"
+	if(F)
+		if(F.on)
+			state += "-flight-on" //"helmet-flight-on" // "helmet-cam-flight-on"
+		else
+			state += "-flight" //etc.
+
+	icon_state = state
+
+	usr.update_inv_head(0)

```

Comment:
After whacking my brain it seems that the commented out code for Sec Berets in the job/security file somehow was screwing with this as well, hopefully I've fixed this.